var/spack/repos/builtin/packages/libsonata/package.py

```diff
@@ -26,20 +28,32 @@ class Libsonata(CMakePackage):
     depends_on('highfive+mpi', when='+mpi')
     depends_on('highfive~mpi', when='~mpi')
     depends_on('mpi', when='+mpi')
+
+    depends_on('python', type=('build', 'run'), when='+python')
     depends_on('py-pybind11@2.0:', type='build', when='+python')
 
     def cmake_args(self):
         result = [
             '-DEXTLIB_FROM_SUBMODULES=OFF',
         ]
         if self.spec.satisfies('+python'):
+            pyspec = self.spec['python']
             result.extend([
                 '-DSONATA_PYTHON=ON',
-                '-DPYTHON_EXECUTABLE:FILEPATH={}'.format(self.spec['python'].command.path),
+                '-DPYTHON_INSTALL_SUFFIX={}'.format(pyspec.package.site_packages_dir),
+                '-DPYTHON_EXECUTABLE:FILEPATH={}'.format(pyspec.command.path),
             ])
         if self.spec.satisfies('+mpi'):
             result.extend([
                 '-DCMAKE_C_COMPILER:STRING={}'.format(self.spec['mpi'].mpicc),
                 '-DCMAKE_CXX_COMPILER:STRING={}'.format(self.spec['mpi'].mpicxx),
             ])
         return result
+
+    def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
+        if self.spec.satisfies("+python"):
+            pyspec = self.spec['python']
+            run_env.prepend_path(
```

Comment:
I saw different ways to put compiled Python modules to Spack prefix in the package repo: some packages put it directly to prefix root (`steps`, `shortbred`), other to `lib` (`root`), other to `lib` and / or `lib64` (`neuron`); and yet other to `site-packages`, but specifying it directly rather than taking from `spec['python']` (`ngttp2`).

I've opted for `site-packages` taken from `spec['python']` as it makes apparent which Python version a given package build is compiled against.
Is there any convention I'd better follow instead?

Reply:
> Is there any convention I'd better follow instead?

At least I am not aware of...I have seen the issue as different packages handle it differently (on one platform in lib and other lib64).

The way you did looks reasonable to me.